src/disputeManager/DisputeManager.ts

```diff
@@ -80,6 +80,25 @@ class DisputeManager {
                 fraudProofsToApply
             } = await this.constructDispute(forkId);
 
+            // Check if there's anything meaningful to dispute
+            const hasTimeout =
+                dispute.input.timeout.participant !== ethers.ZeroAddress;
+            const hasFraudProofs = fraudProofsToApply.length > 0;
+            const hasSelfRemoval = dispute.input.selfRemoval;
+
+            if (!hasTimeout && !hasFraudProofs && !hasSelfRemoval) {
+                this.logger.verbose(
+                    "No timeout, fraud proofs, or self-removal to submit - skipping dispute",
+                    {
+                        forkId,
+                        hasTimeout,
+                        hasFraudProofs,
+                        hasSelfRemoval
+                    }
+                );
+                return;
+            }
+
```

Comment:
The question we need to ask is why and where is dispute being initiated if there are no slashes, no timeout, no self removal and (joining/force processing of an L1 message, but that TS code is yet to be added - hopefully this week just thinking about some stuff)